### PR TITLE
validation improvements

### PR DESCRIPTION
**Pull request by mwhudson**
This contains most of the validation fixes I had in mind, it doesn't yet address the ssh key field, that can wait until tomorrow.

Best to play around in dry run mode I think, reading the code or watching me do it in asciinema probably doesn't convey the sense of this very well :)

### CODE REVIEW

**Comment by mwhudson** [comment]
[validation improvements](https://trello.com/c/Q9PwMi8D/97-validation-improvements)

**Comment by mwhudson** [comment]
This is definitely a better experience than what is currently in master so merging. Flames away!

File: subiquity/ui/views/identity.py [hunk 1]
```diff
@@ -39,10 +48,92 @@
...
+        if len(ch) == 1 and ch in ':,=':
+            self.bff.in_error = True
+            self.bff.show_extra(("info_error", "The characters : , and = are not permitted in this field"))
+            return False
+        else:
+            return super().valid_char(ch)
+
+class UsernameEditor(StringEditor, WantsToKnowFormField):
+    def valid_char(self, ch):
+        if len(ch) == 1 and not re.match('[a-z0-9_-]', ch):
```
Comment (xnox) [inline subiquity/ui/views/identity.py]: I thought username may not start with a digit.

File: subiquity/ui/views/identity.py [hunk 2]
```diff
@@ -39,10 +48,92 @@
...
 
+class SSHImport(WidgetWrap, WantsToKnowFormField):
+
+    signals = ['change']
+
+    _helps = {
+        None: _("You can import your SSH keys from Github, Launchpad or Ubuntu One."),
+        "gh": _("Enter your github username."),
+        "lp": _("Enter your Launchpad username."),
+        "sso": _("Enter an email address associated with your Ubuntu One account."),
```
Comment (xnox) [inline subiquity/ui/views/identity.py]: interesting, did not know this is possible.

File: subiquity/ui/views/identity.py (hunk 1 above)
Comment (xnox) [reply subiquity/ui/views/identity.py]: Oh, but i see there is other validation for first char to be alphabetish, good.

**Pull request merged**